lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -2790,6 +2790,9 @@ void LocalAllocOp::getEffects(
     return;
   effects.emplace_back(MemoryEffects::Allocate::get(),
                        mlir::triton::gpu::SharedMemory::get());
+  if (getSrc())
```

Comment:
I don't believe this is necessary for the PR, but seems like it should be correct.